app/controllers/spree/api/v2/storefront/account/product_questions_answers_controller.rb

```diff
@@ -8,25 +8,21 @@ class ProductQuestionsAnswersController < ::Spree::Api::V2::ResourceController
 
             # GET /api/v2/storefront/account/product_questions_answers
             def index
-              render_serialized_payload { serialize_collection(resource) }
+              render_serialized_payload { serialize_collection(paginated_collection) }
             end
 
             private
 
             def resource
-              resource_finder.user_product_questions(spree_current_user.id)
+              @resource ||= resource_finder.user_product_questions(spree_current_user.id)
             end
 
             def collection_serializer
               Spree::V2::Storefront::ProductQuestionSerializer
             end
 
-            def serialize_collection(collection)
-              collection_serializer.new(
-                collection,
-                include: resource_includes,
-                fields: sparse_fields
-              ).serializable_hash
+            def paginated_collection
+              collection_paginator.new(resource, params).call
```

Comment:
@mausamp we cannot remove resource includes..
that will make your query ineffective as we won't be able to fetch right depth of data..